### PR TITLE
18/hooks tests

### PR DESCRIPTION
**Pull request by pyphilia**
ref #18

### CODE REVIEW

File: src/mutations/item.ts [hunk 1]
```diff
@@ -433,19 +438,22 @@ export default (queryClient: QueryClient, queryConfig: QueryClientConfig) => {
             item.set(
               'path',
               buildPath({
-                prefix: queryClient.getQueryData<Record<Item>>(buildItemKey(to))?.get('path') ?? '',
+                prefix:
+                  queryClient
```
Comment (juancarlosfarah) [inline src/mutations/item.ts:442]: Can we use the default for `get()`, i.e. `get('key', defaultValue)`?

Comment (juancarlosfarah) [reply src/mutations/item.ts:442]: Instead of `??`

File: src/mutations/item.ts [hunk 2]
```diff
@@ -352,7 +354,10 @@ export default (queryClient: QueryClient, queryConfig: QueryClientConfig) => {
               item.set(
                 'path',
                 buildPath({
-                  prefix: queryClient.getQueryData<Record<Item>>(buildItemKey(to))?.get('path') ?? '',
+                  prefix:
+                    queryClient
+                      .getQueryData<Record<Item>>(buildItemKey(to))
```
Comment (juancarlosfarah) [inline src/mutations/item.ts:359]: Same. as below.

Comment (juancarlosfarah) [reply src/mutations/item.ts:359]: Also, can we not factor out the `buildItemKey(to)`?

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Looks very good! Just some minor minor changes. 👍

File: src/mutations/item.ts (hunk 1 above)
Comment (pyphilia) [reply src/mutations/item.ts:442]: Actually  this trick (default value using `get`) works only if the key does not exist, so it's still possible to have an `undefined` value. Related to that, typescript requires a default value, so I would anyway need `?? ''`

**Pull request merged**